### PR TITLE
Introduce gpustat package

### PR DESCRIPTION
**Pull request by wookayin**
I plan to switch to using a package, say `gpustat`, instead of writing everything in a single file `gpustat.py`. 

As a lot of features are going to to added, I feel the necessity of doing this. In this way, we can also add (optional) features such as monitoring, web support, etc.

@Stonesjtu How do you think of this?

### CODE REVIEW

**Comment by Stonesjtu** [comment]
Can't agree more. The current main file has too many lines, and the single file executing makes little sense as dependencies are introduced in previous PR.

File: gpustat/__init__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+"""
+The gpustat module.
+"""
+
+__version__ = '0.5.0.dev1'
+
+from .gpustat import GPUStat, GPUStatCollection
```
[the remark this answers is not in the document]
Comment (wookayin) [reply gpustat/__init__.py]: `gpustat.gpustat` sounds funny, but I have no better naming. Having two files separately for those classes sounds too excessive, but agree with having print-related stuffs in a separate one.

[the remark this answers is not in the document]
Comment (Stonesjtu) [reply gpustat/__init__.py]: When you have a package, you don't need the `gpustat.py`. A suggestion of file structure:
- gpu_stat_collection.py: classes `GPUStat` and `GPUStatCollection`
- display.py output.py or utils.py : the print-related stuffs (There are many lines related to that)
- main.py: the main function (currently `.gpustat.main`)

A reference: https://github.com/nicolargo/glances/

[the remark this answers is not in the document]
Comment (wookayin) [reply gpustat/__init__.py]: Well, it is too verbose, and actually I am looking for something simple that can nicely embrace both.

**Comment by wookayin** [comment]
So we have `core.py` that implements the two classes, and `__main__.py` for some main-related stuffs (so that one can run via `gpustat -m` as well). Will merge soon.

**Comment by wookayin** [comment]
This is now on master.

**Pull request merged**